### ISSUE TITLE
reproducible builds

### ISSUE THREAD

**Issue by leggewie** [freelan-developers/freelan#136]
please have a look at https://gist.githubusercontent.com/lamby/ecf8a75b06aab0b91b519b6157422288/raw from https://tests.reproducible-builds.org/debian/rb-pkg/unstable/amd64/freelan.html

**Comment by s-vincent** [comment]
For the section `objdump --line-numbers --disassemble --demangle --section=.text {}`, it seems that the binary has not been stripped.

**Comment by ereOn** [comment]
I'm not sure I get exactly what is asked of me here. I may be lacking context to fully understand how I can help and what I can answer.

**Comment by leggewie** [comment]
+-- [ https://wiki.debian.org/ReproducibleBuilds/About ]
|
| With free software, anyone can inspect the source code for malicious flaws. But Debian provide
| binary packages to its users. The idea of “deterministic” or “reproducible” builds is to empower
| anyone to verify that no flaws have been introduced during the build process by reproducing
| byte-for-byte identical binary packages from a given source. 
|
+--

One common way to have unreproducible builds (and apparently the one at play here) is when time-related data enters the binary package.  Here it seems to be the time-zone of the buildhost.  https://wiki.debian.org/ReproducibleBuilds/Howto

**Comment by ereOn** [comment]
So I should get rid of the time of build in the binary ? I don't mind if that's the case.

**Comment by leggewie** [comment]
going by https://tests.reproducible-builds.org/debian/rb-pkg/unstable/amd64/freelan.html it seems there might be a few other things.  I haven't tested yet.

**Comment by s-vincent** [comment]
@leggewie do you include the timestamp patch you mentionned in this reproducible build ?

For the GNU_BUILD_ID, I found an interresting link: https://blog.beuc.net/posts/Practical_basics_of_reproducible_builds/. In summary as Debian binary is always build with debug (-g option) then stripped, it differs because build ID is computed from non-stripped executable.

**Comment by leggewie** [comment]
No, I haven't done anything about it, yet.  Except bring it on the radar.  It is a valid goal, but not the most important thing in the world.

**Comment by s-vincent** [comment]
I will commit the patch from Chris Lamb that uses SOURCE_DATE_EPOCH if available (and classic build time if not).

**Comment by leggewie** [comment]
Please do, but leave time to confirm that this issue is fixed entirely before closing the ticket.  There were comments that other changes will be needed.

**Comment by s-vincent** [comment]
@leggewie When Freelan 2.1 debian package will be out, it will be interresting if it still raises errors/warnings in reproducible build.

**Comment by s-vincent** [comment]
@leggewie : Freelan 2.1 has been tagged. Is it possible to push it to Debian and see if it still raises issues ?

**Comment by s-vincent** [comment]
@leggewie Just release FreeLAN 2.2. It would be interresting to see if reproducible builds still raises errors.

**Comment by s-vincent** [comment]
Freelan 2.2-3 debian package is reproducible for amd64 and arm64.
i386 builds has not been triggerred yet.
